### PR TITLE
Socket.IO/Express Set up

### PR DESCRIPTION
**Pull request by billxsheng**
- [ ] Set up Express server
- [ ] Set up Socket.IO wrapper for Express and React.js
- [ ] Added basic functionality in client and server for team demo

**PR Contents**
Added Socket.IO skeleton code where team can extend. 

**Areas of Inspection** 
- Ensure skeleton code is clear and easy to extend
- Ensure client code does not conflict with existing PR's

### CODE REVIEW

**Review by dlhxu — APPROVED** [review]
Looks good to me

**Review by varoongupta — APPROVED** [review]
looks good

**Review by jeffreyquiambao — APPROVED** [review]
lgtm! :)

File: server/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+yarn.lock
+node_modules
```
Comment (rosakng) [inline server/.gitignore:2]: 👍

File: server/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        callback();
+    });
+});
+
+app.use(router);
+
+server.listen(PORT, () => {
+    console.log('Server running on ' + PORT);
+});
```
Comment (rosakng) [inline server/index.js:42]: new line pls 🥺

**Review by rosakng — APPROVED** [review]

**Pull request merged**